From 924a199b73a3b589dddd799ec9771f1ebbde9526 Mon Sep 17 00:00:00 2001
From: Lukas Bulwahn <lukas.bulwahn@gmail.com>
Date: Sun, 12 May 2019 19:09:28 +0200
Subject: [PATCH 080/171] fs/exec: annotate install_exec_creds

HANDLES:

fs/exec.c:1471:2: warning: releasing mutex 'get_current().signal->cred_guard_mutex' that was not held [-Wthread-safety-analysis]
        mutex_unlock(&current->signal->cred_guard_mutex);
        ^
---
 fs/exec.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/exec.c b/fs/exec.c
index 8c63d2f15127..b6f075df9004 100644
--- a/fs/exec.c
+++ b/fs/exec.c
@@ -1447,7 +1447,7 @@ EXPORT_SYMBOL(bprm_change_interp);
 /*
  * install the new credentials for this executable
  */
-void install_exec_creds(struct linux_binprm *bprm)
+void install_exec_creds(struct linux_binprm *bprm) __releases_mutex(current->signal->cred_guard_mutex)
 {
 	security_bprm_committing_creds(bprm);
 
-- 
2.17.1

